src/api/middleware/isAuthenticated.ts

```diff
@@ -0,0 +1,15 @@
+import { Request, Response } from 'express';
+import { Authentication } from '../../domain/model/authentication/Authentication';
+import { ApiError } from '../ApiError';
+
+export async function isAuthenticated(
+  req: Request,
+  res: Response,
+  next: () => any
+) {
+  const authentication = Reflect.get(req, 'authentication') as Authentication;
```

Comment:
Not familiar with this syntax, why do you need the 'as'?  Is there a simpler way to express it?

Reply:
Basically here the typescript compiler has no way of knowing what is the type of the 'authentication' property in the request because that field is not defined in the Request class. The `as Authentication` doesn't provide any runtime safety. It just makes the following code easier to read and safer. 

One better way (though more involved) would be to extend the Request object with an `AuthenticatedRequest` and add the additional parameter. I'll give it a go.